tests/unit/commandlog.tcl

```diff
@@ -0,0 +1,361 @@
+        r config set commandlog-slow-execution-max-len 100
+        r config set commandlog-execution-slower-than 0
+        r commandlog reset slow
+        catch {r acl setuser "slowlog test user" +get +set} _
+        r config set primaryuser ""
+        r config set primaryauth ""
+        r config set requirepass ""
+        r config set tls-key-file-pass ""
+        r config set tls-client-key-file-pass ""
+        r acl setuser slowlog-test-user +get +set
+        r acl getuser slowlog-test-user
+        r acl deluser slowlog-test-user non-existing-user
+        r config set commandlog-execution-slower-than 0
+        r config set commandlog-execution-slower-than -1
+        set slowlog_resp [r commandlog get -1 slow]
+
+        # Make sure normal configs work, but the two sensitive
+        # commands are omitted or redacted
+        assert_equal 11 [llength $slowlog_resp]
+        assert_equal {commandlog reset slow} [lindex [lindex $slowlog_resp 10] 3]
+        assert_equal {acl setuser (redacted) (redacted) (redacted)} [lindex [lindex $slowlog_resp 9] 3]
+        assert_equal {config set primaryuser (redacted)} [lindex [lindex $slowlog_resp 8] 3]
+        assert_equal {config set primaryauth (redacted)} [lindex [lindex $slowlog_resp 7] 3]
+        assert_equal {config set requirepass (redacted)} [lindex [lindex $slowlog_resp 6] 3]
+        assert_equal {config set tls-key-file-pass (redacted)} [lindex [lindex $slowlog_resp 5] 3]
+        assert_equal {config set tls-client-key-file-pass (redacted)} [lindex [lindex $slowlog_resp 4] 3]
+        assert_equal {acl setuser (redacted) (redacted) (redacted)} [lindex [lindex $slowlog_resp 3] 3]
+        assert_equal {acl getuser (redacted)} [lindex [lindex $slowlog_resp 2] 3]
+        assert_equal {acl deluser (redacted) (redacted)} [lindex [lindex $slowlog_resp 1] 3]
+        assert_equal {config set commandlog-execution-slower-than 0} [lindex [lindex $slowlog_resp 0] 3]
+    } {} {needs:repl}
+
+    test {COMMANDLOG slow - Some commands can redact sensitive fields} {
+        r config set commandlog-execution-slower-than 0
+        r commandlog reset slow
+        r migrate [srv 0 host] [srv 0 port] key 9 5000
+        r migrate [srv 0 host] [srv 0 port] key 9 5000 AUTH user
+        r migrate [srv 0 host] [srv 0 port] key 9 5000 AUTH2 user password
+        r config set commandlog-execution-slower-than -1
+        set slowlog_resp [r commandlog get -1 slow]
+
+        # Make sure all 3 commands were logged, but the sensitive fields are omitted
+        assert_equal 4 [llength $slowlog_resp]
+        assert_match {* key 9 5000} [lindex [lindex $slowlog_resp 2] 3]
+        assert_match {* key 9 5000 AUTH (redacted)} [lindex [lindex $slowlog_resp 1] 3]
+        assert_match {* key 9 5000 AUTH2 (redacted) (redacted)} [lindex [lindex $slowlog_resp 0] 3]
+    } {} {needs:repl}
+
+    test {COMMANDLOG slow - Rewritten commands are logged as their original command} {
```

Comment:
This is effectively a duplicate of the version in slowlog.tcl, can we delete that other file and just make this test for both command log and slow log?

Reply:
in theory we can delete slowlog.tcl, but i'd like to keep it in case something different between slowlog and commandlog.